### PR TITLE
N8n 2284 update credentials modal

### PR DESCRIPTION
**Pull request by mutdmour**
borrows from
https://github.com/n8n-io/n8n/pull/2163 to add credential testing
and 
https://github.com/n8n-io/n8n/pull/2029 to set credential names

### CODE REVIEW

**Review by mutdmour — COMMENTED** [review]
hints

File: packages/editor-ui/src/components/helpers.ts [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+export function convertToDisplayDate (epochTime: number) {
+	return dateformat(epochTime, 'yyyy-mm-dd HH:MM:ss');
+}
+
+export function convertToHumanReadableDate (epochTime: number) {
+	return dateformat(epochTime, 'd mmmm, yyyy @ HH:MM Z');
+}
+
+export function getAppNameFromCredType(name: string) {
+	return name.split(' ').filter((word: string) => !KEYWORDS_TO_FILTER.includes(word)).join(' ');
```
Comment (ivov) [inline packages/editor-ui/src/components/helpers.ts:20]:
```suggestion
	return name.split(' ').filter((word) => !KEYWORDS_TO_FILTER.includes(word)).join(' ');
```

File: packages/editor-ui/src/components/helpers.ts [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
 		? html.replace(/href=/g, 'target="_blank" href=')
 		: html;
 }
+
+export function convertToDisplayDate (epochTime: number) {
+	return dateformat(epochTime, 'yyyy-mm-dd HH:MM:ss');
+}
+
+export function convertToHumanReadableDate (epochTime: number) {
+	return dateformat(epochTime, 'd mmmm, yyyy @ HH:MM Z');
```
Comment (ivov) [inline packages/editor-ui/src/components/helpers.ts:16]: Unlike American English style, I think we do not use a comma between month and year when the date starts with the day.

File: packages/editor-ui/src/components/mixins/showMessage.ts [hunk 3]
```diff
@@ -81,6 +83,23 @@ export const showMessage = mixins(externalHooks).extend({
 			});
 		},
 
+		async confirmMessage (message: string, headline: string, type = 'warning' as MessageType | null, confirmButtonText = 'OK', cancelButtonText = 'Cancel'): Promise<boolean> {
```
Comment (ivov) [inline packages/editor-ui/src/components/mixins/showMessage.ts:86]: `type: MessageType | null = 'warning'`

File: packages/editor-ui/src/components/mixins/showMessage.ts [hunk 4]
```diff
@@ -81,6 +83,23 @@ export const showMessage = mixins(externalHooks).extend({
...
+			try {
+				const options: ElMessageBoxOptions  = {
+					confirmButtonText,
+					cancelButtonText,
+					dangerouslyUseHTMLString: true,
+				};
+				if (type) {
+					options.type = type;
+				}
```
Comment (ivov) [inline packages/editor-ui/src/components/mixins/showMessage.ts:95]: Another option:
```suggestion
				const options: ElMessageBoxOptions  = {
					confirmButtonText,
					cancelButtonText,
					dangerouslyUseHTMLString: true,
					...(type && { type }),
				};
```

File: packages/editor-ui/src/modules/credentials.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+			let newName = DEFAULT_CREDENTIAL_NAME;
+			if (!TYPES_WITH_DEFAULT_NAME.includes(credentialTypeName)) {
+				const { displayName } = context.getters.getCredentialTypeByName(credentialTypeName);
+				newName = getAppNameFromCredType(displayName);
+				newName = newName.length > 0 ? `${newName} ${DEFAULT_CREDENTIAL_POSTFIX}` : DEFAULT_CREDENTIAL_NAME;
+			}
+
+			try {
+				const res = await getCredentialsNewName(context.rootGetters.getRestApiContext, newName);
+				newName = res.name;
```
Comment (ivov) [inline packages/editor-ui/src/modules/credentials.ts:171]: Another option:
```suggestion
				return await getCredentialsNewName(context.rootGetters.getRestApiContext, newName).then(res => res.name);
```

File: packages/cli/src/Server.ts [hunk 6]
```diff
@@ -1594,6 +1594,44 @@ class App {
 			),
 		);
 
+		this.app.get(
+			[`/${this.restEndpoint}/credential-icon/:credentialType`],
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/cli/src/Server.ts:1598]: oh this is a copy/paste artifact.. will update

File: packages/design-system/src/components/N8nInfoTip/InfoTip.stories.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+};
+
+const Template = (args, { argTypes }) => ({
+	props: Object.keys(argTypes),
+	components: {
+		N8nInfoTip,
+	},
+	template:
+		'<n8n-info-tip>Need help doing something? <a href="/docs" target="_blank">Open docs</a></n8n-info-tip>',
+});
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/design-system/src/components/N8nInfoTip/InfoTip.stories.js:15]: would rather add it when it's used..

File: packages/editor-ui/src/components/CredentialEdit/CredentialConfig.vue [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+			message="Please check the errors below"
+		/>
+
+		<banner
+			v-if="authError && !showValidationWarning"
+			theme="danger"
+			message="Couldn’t connect with these settings."
+			:details="authError"
+			buttonLabel="Retry"
+			buttonLoadingLabel="Retrying"
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/components/CredentialEdit/CredentialConfig.vue:15]: not really needed..

File: packages/editor-ui/src/components/CredentialEdit/CredentialConfig.vue [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+		documentationUrl(): string {
+			const type = this.credentialType as ICredentialType;
+
+			if (!type) {
+				return '';
+			}
+
+			if (type.documentationUrl && type.documentationUrl.startsWith('http')) {
+				return type.documentationUrl;
+			}
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/components/CredentialEdit/CredentialConfig.vue]: nice

File: packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue [hunk 10]
```diff
@@ -0,0 +1,853 @@
...
+
+				return;
+			}
+
+			const params = `scrollbars=no,resizable=yes,status=no,titlebar=noe,location=no,toolbar=no,menubar=no,width=500,height=700`;
+			const oauthPopup = window.open(url, 'OAuth2 Authorization', params);
+			Vue.set(this.credentialData, 'oauthTokenData', null);
+
+			const receiveMessage = (event: MessageEvent) => {
+				// // TODO: Add check that it came from n8n
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue:776]: legacy thing I don't understand completely.. we can worry about this later

File: packages/editor-ui/src/components/mixins/showMessage.ts (hunk 4 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/mixins/showMessage.ts:95]: nice

File: packages/editor-ui/src/modules/credentials.ts (hunk 5 above)
Comment (mutdmour) [reply packages/editor-ui/src/modules/credentials.ts:171]: I prefer not to use `.then` for clarity but I will refactor this a bit.

File: packages/editor-ui/src/plugins/compontents.ts [hunk 11]
```diff
@@ -10,9 +10,7 @@ import Dialog from 'element-ui/lib/dialog';
 import Dropdown from 'element-ui/lib/dropdown';
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/plugins/compontents.ts:10]: nice catch

File: packages/editor-ui/src/components/helpers.ts (hunk 2 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/helpers.ts:16]: will discuss with product

**Review by ivov — APPROVED** [review]
Thank you for the great work

**Comment by janober** [comment]
Really great. Thanks a lot. Got merged.

File: packages/design-system/theme/src/common/popup.scss [hunk 12]
```diff
@@ -31,8 +35,8 @@
 	top: 0;
 	width: 100%;
 	height: 100%;
-	opacity: var.$popup-modal-opacity;
-	background: var.$popup-modal-background-color;
+	background-color: var.$popup-modal-background-color;
+	backdrop-filter: blur(4px) opacity(1);
```
[the remark this answers is not in the document]
Comment (umutbozdag) [reply packages/design-system/theme/src/common/popup.scss:39]: Hi!

Actually you could use a workaround here to make `background-filter` property work in the Firefox.

You could use `@supports` at-rule to check if the browser supports that property and then you would have different ways to solve this. Like this:

```
@supports (-webkit-backdrop-filter: none) or (backdrop-filter: none) {
  .blurred-container {
    -webkit-backdrop-filter: blur(10px) opacity(1);
    backdrop-filter: blur(10px) opacity(1);
  }
}

@supports not ((-webkit-backdrop-filter: none) or (backdrop-filter: none)) {
  .blurred-container {
    background-color: rgba(255, 255, 255, .8); // Relevant CSS property comes here.
  }
}
```

And [this](https://stackoverflow.com/a/20039965/10991790) source can help you too.

[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/design-system/theme/src/common/popup.scss:39]: thanks @umutbozdag for sharing.. will take a look

**Pull request merged**